lib/submit-analytics.js

```diff
@@ -33,50 +104,184 @@ function sha256(data) {
     return hash.digest('hex');
 }
 
-if (isAnalyticsDisabled()) {
-    module.exports = function(){};
-} else {
+/**
+ * Send collected analytics data to Realm's servers over HTTPS
+ * @param  {[webhookProdDetails|webhookStageDetails|mixpanelDetails]} requestFactory Submission URL generator
+ * @param  {Object} payload analytics info
+ */
+async function dispatchAnalytics(requestFactory, payload) {
+    const https = require('https');
+
+    return new Promise((resolve, reject) => {
+        const webhookRequest = new requestFactory().buildRequest(payload);
+        var req = https.get(webhookRequest, function (result) {
+            result.on('data', (data) => {
+                // we don't log the data that is returned -- we're just interested in the header
+            });
+        });
+
+        req.on('response', response => {
+            response.on('end', () => {
+                resolve(
+                    {
+                        statusCode:  response.statusCode,
+                        statusMessage: response.statusMessage,
+                        errorMessage: '',
+                    });
+                return;
+            });
+
+            response.on('error', error => {
+                reject(
+                    {
+                        statusCode:  response.statusCode,
+                        statusMessage: response.statusMessage,
+                        errorMessage: error,
+                    });
+                return;
+            });
+        });
+    });
+}
+
+/**
+ * Collect analytics data from the runtime system
+ * @param  {Object} context contents of the app's package.json file
+ * @param  {string} eventName Custom tag that signifies what the app was doing when the analytics were submitted
+ * @returns {Object} Analytics payload
+ */
+function fetchPlatformData(context, eventName) {
     const os = require('os');
-    const request = require('request');
     const { machineId } = require("node-machine-id");
+    const environment = utils.getContext();
+
+    return new Promise((resolve, reject) => {
+        machineId()
+        .catch((e) => {
+            reject('Analytics submission failed');
+            return;
+        })
+
+        .then((identifier) => {
+            if (!identifier) {
+                identifier = sha256('unknown');
+            }
 
-    module.exports = function(eventName, context) {
-        // Submit analytics after some time - we do this to give the application a chance to disable analytics.
+            // payloads for webhook and MixPanel differ slightly
+            const payload = {
```

Comment:
```suggestion
            const payloads = {
```

It feels more appropriate to pluralize this, since we're basically building two separate objects here.

Reply:
+1